docs/core/howto/logger.rst

```diff
@@ -101,37 +101,50 @@ For example:
 This example will show the string "object with value 7 doing a task" because the ``log_source`` key is automatically set to the ``MyObject`` instance that the ``Logger`` is retrieved from.
 
 
-Capturing Failures
-~~~~~~~~~~~~~~~~~~
+Handling Failures
+~~~~~~~~~~~~~~~~~
 
-:py:class:`Logger <twisted.logger.Logger>` provides a :py:meth:`failure <twisted.logger.Logger.failure>` method, which allows one to capture a :py:class:`Failure <twisted.python.failure.Failure>` object conveniently:
+:py:class:`Logger <twisted.logger.Logger>` provides a :py:meth:`failure <twisted.logger.Logger.handlingFailures>` method, which allows one to run some application code, then capture a :py:class:`Failure <twisted.python.failure.Failure>` in the log stream if that code raises an exception:
 
 .. code-block:: python
 
     from twisted.logger import Logger
     log = Logger()
 
-    try:
+    with log.handlingFailures("While doing some math:"):
```

Comment:
Just a suggestion.
To make the name closer to the intended usage

> Note that this API is meant to capture **unexpected and otherwise unhandled** errors (in other words: bugs, which is why tracebacks are preserved).

```suggestion
    with log.handleUnexpectedError("While doing some math:"):
```

Reply:
I thought for a really long time about this, and I think I am going to reject the suggestion on the basis of length; this is already pretty wordy, and we put all the words in to explain its purpose (`with log.handlingUnexpectedErrorsWithMessageAndContinuing`) it gets unwieldy, and "unexpected" is a slippery word to put in a place as definitive as a name. (After all, we *kinda* expect it here, it's the application code which doesn't expect it, and "application" is an ambiguous word, and `ErrorsUnexpectedByApplication`...)

However, with so much thinking about the name, I did realize that this does not scan in an english-readability way. I would not say "with log handling failures as F" in english, I'd say "with failures handled as"; this has the additional ergonomic feature that auto-completion tools will present `log.failure` and `log.failuresHandledAs` next to each other, so I am going to change the name after all.